### ISSUE TITLE
JobLocalBeginMsg messages are not always matched by JobLocalDoneMsg messages

### ISSUE THREAD

**Issue by mich181189** [icecc/icecream#282]
This comes out of what I initially  thought was an icemon bug: https://github.com/icecc/icemon/issues/33

It seems https://github.com/icecc/icecream/commit/1c15f6b9c6ddd329e4fe1e2a03c7420a0407ae25 added some JobLocalBeginMsg messages on preprocessing to account for preprocessing load. It does not, however, add any JobLocalDoneMsg messages to match. This manifests itself as a very wrong view in icemon, and possibly bad calculations about system load,

**Comment by mich181189** [comment]
I tried adding JobLocalDoneMsg messages to match, but it didn't have the desired effect in icemon and also caused problems with it causing an M_END message, which then caused one of the sockets to disconnect.

Reverting this commit definitely seems to fix it though.

**Comment by HenryMiller1** [comment]
We should be able to change our tests:

Create a fake icecream monitor that tracks messages.
After every test completes check the monitor to verify that the count JobLocalBeginMsg matches the count of JobLocalEndMsg. In tests/test.sh look for uses of flush_logs/check_logs_for_generic_errors (there are several other check_logs for errors functions) for places to inject this check.

I believe this test is easy to create so I don't want to call this fixed until we have the check, even if this manifestation of the bug is fixed.

**Comment by mich181189** [comment]
I'd agree about testing.

TL;DR: I think the precompile tracking needs more thought

I think the best plan might be to revert the commit - the only way JobLocalDoneMsg messages are normally raised is by the client exiting. This means the daemon is not expecting to receive them from the client. It's a bit of a rabbit hole.

In fact, when the daemon receives a JoblocalBeginMsg from the client, it assumes it's in the linking phase - without changing the protocol to indicate explicitly this is a preprocessor job I'm not sure how this could really be fixed. 

If you want to have tracking of preprocessor jobs, I'm beginning to think adding a new message type would be the way to go - JobLocalBeginMsg seems to expect to be a linker job.

**Comment by danny-smit** [comment]
Reverting the commit of 1c15f6b seems to solve the problem of issue #280 as well. Is reverting that a reasonable way to go for the 1.1 release? And giving the concept about precompiling more thought for a next release?

**Comment by HenryMiller1** [comment]
1c15f6b9c6ddd329e4fe1e2a03c7420a0407ae25 reverted
